app/domain/authentication/authn_gcp/restrictions_validator.rb

```diff
@@ -0,0 +1,105 @@
+module Authentication
+  module AuthnGCP
+
+    # This class is responsible of validating resource restrictions combination.
+    # the allowed combination is: Role should have at least 1 permitted constraints, and non of illegal constraints
+    class RestrictionsValidator
+
+      ILLEGAL_STATE_MISSING_CONSTRAINTS_MESSAGE = "constraints are missing or empty"
+      ILLEGAL_STATE_MISSING_RESOURCE_TYPE_MESSAGE = "resource restriction type is missing or empty"
+      ILLEGAL_STATE_DUPLICATE_CONSTRAINTS_MESSAGE = "constraints contain duplications"
+      ILLEGAL_STATE_DUPLICATE_RESTRICTIONS_MESSAGE = "resource restrictions contain duplications"
+
+      def initialize(resource_restrictions:, permitted_constraints:, logger:)
+        @resource_restrictions = resource_restrictions
+        @permitted_constraints = permitted_constraints
+        @logger = logger
+
+        @logger.debug(LogMessages::Authentication::AuthnGCP::ValidatingRestrictionsConstraintCombination.new)
+        validate
+        @logger.debug(LogMessages::Authentication::AuthnGCP::ValidatedRestrictionsConstraintCombination.new)
+      end
+
+      private
+
+      def validate
+        input_validation
+        validate_constraints_are_permitted
+        validate_restrictions_values_exist
```

Comment:
why are these methods not in `input_validation`? how are they different than the validations performed there?

btw, i think that we can get rid of the `input_validation` method and move its validations to `validate`.

Reply:
OK i will move them, i though it is too noisy to mix the logic with the parameters validation